### ISSUE TITLE
Recent version of flutter_test not compatible with mockito

### ISSUE THREAD

**Issue by KevinTheGray** [flutter/flutter#20011]
I am currently unable to use Mockito with flutter_test.  If I use `mockito: 3.0.0-beta+1` in my `pubspec.yaml`, I get the following error when running `flutter test`:
```
$ flutter test
Building flutter tool...
Running "flutter packages get" in testmockito...             0.9s
00:04 +0: ... /Users/KG/Developer/Mine/testing_safe_to_delete_all_contents/testmockito/test/widget_test.dart           compiler message: file:///Users/KG/.pub-cache/hosted/pub.dartlang.org/mockito-3.0.0-beta+1/lib/src/call_pair.dart:32:22: Error: The class 'TypeMatcher' is abstract and can't be instantiated.
compiler message:       : call = const TypeMatcher<Invocation>();
compiler message:                      ^^^^^^^^^^^
compiler message: file:///Users/KG/.pub-cache/hosted/pub.dartlang.org/mockito-3.0.0-beta+1/lib/src/call_pair.dart:1: Error: Cannot invoke a non-'const' constructor where a const expression is expected.
compiler message: Try using a constructor or factory that is 'const'.
compiler message: file:///Users/KG/.pub-cache/hosted/pub.dartlang.org/mockito-3.0.0-beta+1/lib/src/call_pair.dart:32:22: Error: Too few positional arguments: 1 required, 0 given.
compiler message:       : call = const TypeMatcher<Invocation>();
compiler message:                      ^
00:04 +0 -1: loading /Users/KG/Developer/Mine/testing_safe_to_delete_all_contents/testmockito/test/widget_test.dart [E]
  Failed to load "/Users/KG/Developer/Mine/testing_safe_to_delete_all_contents/testmockito/test/widget_test.dart":
  Compilation failed
  Test: /Users/KG/Developer/Mine/testing_safe_to_delete_all_contents/testmockito/test/widget_test.dart
  Shell: /Users/KG/Developer/Flutter/flutter/bin/cache/artifacts/engine/darwin-x64/flutter_tester
```

If I use `mockito: 3.0.0-beta+2` in my `pubspec.yaml`, I get the following error when running `flutter test`:
```
Running "flutter packages get" in testmockito...                 
Because every version of flutter_test from sdk depends on test 0.12.41 which depends on matcher >=0.12.2 <0.12.3, every version of flutter_test from sdk requires matcher >=0.12.2 <0.12.3.
And because mockito >=3.0.0-beta+2 depends on matcher ^0.12.3, flutter_test from sdk is incompatible with mockito >=3.0.0-beta+2.
So, because testmockito depends on both flutter_test any from sdk and mockito ^3.0.0-beta+2, version solving failed.
pub get failed (1)
```

I believe the problem is with this file: https://github.com/flutter/flutter/blob/master/packages/flutter_test/pubspec.yaml

which locks matcher to `matcher: 0.12.2+1`.  

## Steps to Reproduce

1. `flutter create` a new project
2.  add `mockito: ^3.0.0-beta+1` under `dev_dependencies:` in `pubspec.yaml`
3.  `import 'package:mockito/mockito.dart';` into the test file that was created in the `test` folder
4. run `flutter test`, observe errors
5. 2.  chance to `mockito: ^3.0.0-beta+2`
6. run `flutter test`, observer errors

## Flutter Doctor
```
[✓] Flutter (Channel master, v0.5.8-pre.179, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.5.8-pre.179 at /Users/KG/Developer/Flutter/flutter
    • Framework revision 711ecf7f58 (3 hours ago), 2018-07-30 12:35:15 -0700
    • Engine revision af855e063a
    • Dart version 2.0.0-dev.69.4.flutter-08f59e5de3

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/KG/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • ANDROID_HOME = /Users/KG/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 26.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Community Edition (version 2017.2.7)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 21.2.2
    • Dart plugin version 172.4343.25

[!] VS Code (version 1.21.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
```

**Comment by zoechi** [comment]
```yml
dev_dependencies:
  mockito: ^3.0.0-beta+2
```
as a workaround should do until the dependency constraints are updated.

**Comment by KevinTheGray** [comment]
@zoechi this works until you run `flutter test` and have a test file that imports mockito.  If you do that, you get the constraint error.

Certain repos request that the tests pass before being able to merge a PR.  e.g https://github.com/flutter/flutter_markdown.  Until these dependency issues are resolved, the tests will fail.

**Comment by KevinTheGray** [comment]
somewhere along the line, this got fixed, and I can use both `^3.0.0-beta+2` and `^3.0.0-beta+1`.  Going to close this.

current flutter doctor
```
[✓] Flutter (Channel master, v0.5.9-pre.64, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.5.9-pre.64 at /Users/KG/Developer/Flutter/flutter
    • Framework revision b5b55447a4 (26 hours ago), 2018-08-16 08:47:20 -0700
    • Engine revision 4ee648914f
    • Dart version 2.1.0-dev.0.0.flutter-be6309690f

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/KG/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 27.0.3
    • ANDROID_HOME = /Users/KG/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 27.1.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Community Edition (version 2017.2.7)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 21.2.2
    • Dart plugin version 172.4343.25

[!] VS Code (version 1.21.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
```
